torch/onnx/_internal/fx/passes/type_promotion.py

```diff
@@ -20,15 +21,26 @@
 )
 from torch._refs import linalg as _linalg_refs, nn as _nn_refs, special as _special_refs
 from torch._refs.nn import functional as _functional_refs
+from torch._subclasses import fake_tensor
+from torch.fx.experimental import proxy_tensor
 
 # Imported to resolve beartype issue when type checking node.Argument.
 from torch.fx.node import Node  # noqa: F401
 
 from torch.onnx._internal import _beartype
-from torch.utils import _pytree
+from torch.onnx._internal.fx import _pass, diagnostics
+from torch.utils import _python_dispatch, _pytree
 
 logger = logging.getLogger(__name__)
 
+# TODO(bowbao): move to type utils.
+_SCALAR_TYPE_TENSOR_DTYPE_MAP: Mapping[type, torch.dtype] = {
```

Comment:
Also please make it accessible through a stateless function or class.

Reply:
Maybe we can unify the need of torch.dtype conversion into something like JitScalarType system now. The needs are increasing.

Reply:
> Also please make it accessible through a stateless function or class.

If it is private I think it's fine for now

Reply:
Should we make function accessor `get(..., None)` style? I'm not sure but it is used here so I'm leaving out type_utils in this PR.